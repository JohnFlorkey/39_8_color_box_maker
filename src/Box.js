import React from 'react';
import './Box.css';

function Box({backgroundColor, width, height, id, removeBox}) {
	return (
		<div className="Box-Container">
			<div className="Box" id={id} style={{backgroundColor: backgroundColor, width: width, height: height}}></div>
			<button onClick={removeBox}>X</button>
		</div>
	);
}

export default Box;