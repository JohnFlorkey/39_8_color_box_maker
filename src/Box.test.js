import { render } from '@testing-library/react';
import Box from './Box';

// smoke test
it("should render without error", () => {
    render(<Box />);
});

// snapshot test
it("should match snapshot", () => {
    const {asFragment} = render(<Box backgroundColor="red" width="200" height="150" />);
    expect(asFragment()).toMatchSnapshot();
})