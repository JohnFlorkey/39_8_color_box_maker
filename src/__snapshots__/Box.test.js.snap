// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box-Container"
  >
    <div
      class="Box"
      style="background-color: red;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
